agent/dns.go

```diff
@@ -1602,21 +1632,24 @@ func (d *DNSServer) handleRecurse(resp dns.ResponseWriter, req *dns.Msg) {
 // resolveCNAME is used to recursively resolve CNAME records
 func (d *DNSServer) resolveCNAME(cfg *dnsConfig, name string, maxRecursionLevel int) []dns.RR {
 	// If the CNAME record points to a Consul address, resolve it internally
-	// Convert query to lowercase because DNS is case insensitive; d.domain is
-	// already converted
-
-	if strings.HasSuffix(strings.ToLower(name), "."+d.domain) {
```

Comment:
I think you can check for the `altDomain` suffix here and keep the rest of this function untouched. Looping over the domains shouldn't be necessary.